### PR TITLE
Upgrade to Angular 14

### PR DESCRIPTION
**Pull request by dlq84**
This PR upgrades to Angular 14. It does pass yarn test and yarn build locally.

It also includes this PR #1781, closed that one because this replaces it.

### CODE REVIEW

File: packages/apollo-angular/package.json [hunk 1]
```diff
@@ -28,16 +28,16 @@
...
     "test": "NODE_OPTIONS=--experimental-modules jest --config jest.config.js",
     "prebuild": "rm -rf build/",
     "deploy": "yarn build && npm publish build",
     "deploy:next": "yarn build && npm publish build --tag next"
   },
   "peerDependencies": {
-    "@angular/core": "^12.0.0 || ^13.0.0",
-    "@apollo/client": "^3.0.0",
+    "@angular/core": "^13.0.0 || ^14.0.0",
+    "@apollo/client": "~3.5.0",
```
Comment (de-dan) [inline packages/apollo-angular/package.json:40]:
```suggestion
    "@apollo/client": "^3.5.0",
```

Comment (dlq84) [reply packages/apollo-angular/package.json]: Thank, this is now fixed

**Comment by dlq84** [comment]
So looks like if you compile with Angular 14 you can't use it in previous versions: https://github.com/angular/angular/issues/46748#issuecomment-1178817658

**Comment by dlq84** [comment]
`@cypress/schematic@2.0.0` changed map/file structures, this is now handled.

**Comment by dlq84** [comment]
@de-dan Is there anything I need to do to get rid of Angular 12 and 13 tests?

**Comment by de-dan** [comment]
I don't know unfortunately. I've seen your changes in `.github/workflows/main.yml` - that should be it.
@kamilkisiela could you have a look?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -88,7 +88,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        angular_version: [12, 13]
+        angular_version: [14]
```
Comment (charlypoly) [inline .github/workflows/main.yml:91]:
```suggestion
        angular_version: [12, 13, 14]
```

**Comment by kamilkisiela** [comment]
Thank you

**Comment by kamilkisiela** [comment]
`4.0.0`

**Pull request merged**